### PR TITLE
fix(Crash): Fix crash in Vehicle::AddPassenger

### PR DESCRIPTION
**Pull request by walkline**
## Changes Proposed:
This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

As I understand it, a try-catch block cannot intercept the ABORT function. Therefore, let's use a simple if statement instead.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15998

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Vehicle/Vehicle.cpp [hunk 1]
```diff
@@ -388,22 +388,20 @@ bool Vehicle::AddPassenger(Unit* unit, int8 seatId)
...
-            LOG_INFO("vehicles", "CRASH! Try-catch in Unit::SetCharmedBy(). Unit {}, typeid: {}, in world: {}, duringremove: {} has wrong CharmType! Charmer {}, typeid: {}, in world: {}, duringremove: {}.", _me->GetName(), _me->GetTypeId(), _me->IsInWorld(), _me->IsDuringRemoveFromWorld(), unit->GetName(), unit->GetTypeId(), unit->IsInWorld(), unit->IsDuringRemoveFromWorld());
+            LOG_INFO("server", "Crash recovered in Unit::SetCharmedBy(). Is: {}!", _me->IsInWorld());
+            LOG_INFO("server", "Crash recovered in Unit::SetCharmedBy(). Is2: {}!", _me->IsDuringRemoveFromWorld());
+            LOG_INFO("server", "Crash recovered in Unit::SetCharmedBy(). Unit {}!", _me->GetName());
+            LOG_INFO("server", "Crash recovered in Unit::SetCharmedBy(). typeid: {}!", _me->GetTypeId());
+            LOG_INFO("server", "Crash recovered in Unit::SetCharmedBy(). Unit {}, typeid: {}, in world: {}, duringremove: {} has wrong CharmType! Charmer {}, typeid: {}, in world: {}, duringremove: {}.", _me->GetName(), _me->GetTypeId(), _me->IsInWorld(), _me->IsDuringRemoveFromWorld(), unit->GetName(), unit->GetTypeId(), unit->IsInWorld(), unit->IsDuringRemoveFromWorld());
```
Comment (Kitzunu) [inline src/server/game/Entities/Vehicle/Vehicle.cpp:404]: keep it vechicles in log

File: src/server/game/Entities/Vehicle/Vehicle.cpp [hunk 2]
```diff
@@ -388,22 +388,20 @@ bool Vehicle::AddPassenger(Unit* unit, int8 seatId)
...
-            if (!_me->SetCharmedBy(unit, CHARM_TYPE_VEHICLE))
-                ABORT();
-        }
-        catch (...)
-        {
-            LOG_INFO("vehicles", "CRASH! Try-catch in Unit::SetCharmedBy()!");
-            LOG_INFO("vehicles", "CRASH! Try-catch in Unit::SetCharmedBy(). not null: {}", _me ? 1 : 0);
+            // I assume SetCharmedBy should always be true.
+            // If not, let's log some debug info.
+            LOG_INFO("server", "Crash recovered in Unit::SetCharmedBy()!");
```
Comment (Kitzunu) [inline src/server/game/Entities/Vehicle/Vehicle.cpp:396]: this first log is redundant as we show the same info in the second one

File: src/server/game/Entities/Vehicle/Vehicle.cpp (hunk 1 above)
Comment (walkline) [reply src/server/game/Entities/Vehicle/Vehicle.cpp:404]: That's fine by me, but I just wanted to make it more visible. I think most users have this log type (vehicle) disabled.

Comment (Kitzunu) [reply src/server/game/Entities/Vehicle/Vehicle.cpp:404]: if it has to do with vehicles, it should be tied to vehicles.

What is output in logs should be easily changed, therefore we need to keep things category specific.

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**